File: 20170126_ConditionalLogic/ConditionalLogic.py
class ConditionalLogic():
    def main(self):
        a = 1
        b = 2
        c = 3
        print("Testing and/or statements")
        print("a = 1\nb = 2\nc = 3")
        print("-----")
        
        testvar = ("Testing '(a == 2 or b == 2)'... ")
        if (a == 2 or b ==2): 
            print(testvar + "One of them is true")
        else: 
            print(testvar + "false")
        
        testvar = ("Testing '(a == 2 and b == 2)'... ")
        if (a == 2 and b ==2):
            print(testvar + "One of them is true")
        else:
            print(testvar + "false")

        testvar = "Testing '(a == 1 or b == 1 and c == 1)'... "
        if (a == 1 or b == 1 and c == 1): 
            print(testvar + "a is true")
        else: 
            print(testvar + "false")

        testvar = "Testing '(a == 1 or b == 1 and c==1)'... "
        if (a==1 or b==1 and c==1):
            print(testvar + "One of them is true")
        else:
            print(testvar + "false")

        # order of precedence test for and
        testvar = "Testing '(3 != b and 9 == c)'... "
        if (3 != b or 9 == c):
            print(testvar + "true")
        else:
            print(testvar + "false")
    
        # order of precedence test for or
        testvar = "Testing '(9 == c or 3 != b)'... "
        if (9 == c or 3 != b):
            print(testvar + "true")
        else:
            print(testvar + "false")



if __name__ == "__main__":
    cl = ConditionalLogic()
    cl.main()

File: 20170126_ConditionalLogic/ConditionalLogic.java
class ConditionalLogic{
  public static void main(String[] args){
    int a = 1, b = 2, c = 3;
    System.out.println("Testing 'and/or' statements'");
    System.out.println("a = 1");
    System.out.println("b = 2");
    System.out.println("c = 3");
    System.out.println("-----");
    // a or b
    System.out.print("Testing '(a==2 || b==2)'... ");
    if (a == 2 || b == 2){
      System.out.println("One of them is true");
    } else{
      System.out.println("false");
    }
    // a and b
    System.out.print("Testing '(a == 2 && b == 2)'... ");
    if (a == 2 && b == 2){
      System.out.println("Both are true");
    }else{
      System.out.println("false");
    }
    // a is true
    System.out.print("Testing '(a == 1 || b == 1 && c==1)'... ");
    if (a == 1 || b==1 && c==1){
      System.out.println("a is true");
    } else{
      System.out.println("false");
    }
    // b and c is true
    System.out.print("Testing '(a == 2 || b == 1 || c == 3)'... ");
    if (a != 2 && b == 2 || c == 2){
      System.out.println("One of them is true");
    } else{
      System.out.println("false");
    }
    // order of precedence test for or
    System.out.print("Testing '((int) 3 != b || (int) 9 == c)'... ");
    if ((int) 3 != b || (int) 9 == c){
      System.out.println("true");
    } else{
      System.out.println("false");
    }
    // order of precedence test for or
    System.out.print("Testing '((int) 9 == c || (int) 3 != b)'... ");
    if ((int) 9 == c || (int) 3 != b){
      System.out.println("true");
    } else{
      System.out.println("false");
    }
  }
}
